### PR TITLE
Bump Astarte device sdk to v1.2.0

### LINKED ISSUE #88: Bump Astarte device sdk to v1.2.0
Bump to latest version of Astarte device sdk.

https://github.com/edgehog-device-manager/edgehog-esp32-device/blob/b7b626035d41d89cceddb97e071a3bd0c2f4ae66/idf_component.yml#L27

### PR DESCRIPTION
**Pull request by harlem88**
The edgehog example binary has grown to `0x1065a0` bytes, and doesn't allow using the default ota partition table, so we add a custom partition table where the smallest app partition is `0x12c000` bytes with `0x25a60` bytes (`13%`) free.

Closes #88 .

### CODE REVIEW

File: src/edgehog_ota.c [hunk 1]
```diff
@@ -196,43 +196,50 @@ void edgehog_ota_init(edgehog_device_handle_t edgehog_dev)
...
+        = astarte_bson_deserializer_element_lookup(doc, "uuid", &req_uuid_element);
+    if (astarte_err != ASTARTE_OK || req_uuid_element.type != BSON_TYPE_STRING) {
         ESP_LOGE(TAG, "Unable to extract requestUUID from bson");
         esp_event_post(EDGEHOG_EVENTS, EDGEHOG_OTA_FAILED_EVENT, NULL, 0, 0);
         return EDGEHOG_ERR_OTA_INVALID_REQUEST;
     }
+
+    uint32_t req_uuid_len;
+    const char *req_uuid
+        = astarte_bson_deserializer_element_to_string(req_uuid_element, &req_uuid_len);
```
Comment (sorru94) [inline src/edgehog_ota.c:216]: `&req_uuid_len` can be replaced by `NULL` if not used.

File: src/edgehog_ota.c [hunk 2]
```diff
@@ -196,43 +196,50 @@ void edgehog_ota_init(edgehog_device_handle_t edgehog_dev)
...
+    astarte_err = astarte_bson_deserializer_element_lookup(doc, "url", &url_element);
+    if (astarte_err != ASTARTE_OK || url_element.type != BSON_TYPE_STRING) {
         ESP_LOGE(TAG, "Unable to extract URL from bson");
         pub_ota_event(
             edgehog_dev, req_uuid, OTA_EVENT_FAILURE, 0, EDGEHOG_ERR_OTA_INVALID_REQUEST, "");
         esp_event_post(EDGEHOG_EVENTS, EDGEHOG_OTA_FAILED_EVENT, NULL, 0, 0);
         return EDGEHOG_ERR_OTA_INVALID_REQUEST;
     }
+    uint32_t ota_url_len;
+    const char *ota_url = astarte_bson_deserializer_element_to_string(url_element, &ota_url_len);
```
Comment (sorru94) [inline src/edgehog_ota.c:229]: `&ota_url_len` can be replaced by `NULL` if not used.

File: src/edgehog_ota.c [hunk 3]
```diff
@@ -557,18 +564,17 @@ static void pub_ota_event(edgehog_device_handle_t edgehog_dev, const char *reque
...
+    astarte_bson_serializer_append_string(bs, "requestUUID", request_uuid);
+    astarte_bson_serializer_append_string(bs, "status", status);
+    astarte_bson_serializer_append_int32(bs, "statusProgress", status_progress);
+    astarte_bson_serializer_append_string(bs, "statusCode", status_code);
+    astarte_bson_serializer_append_string(bs, "message", message);
+    astarte_bson_serializer_append_end_of_document(bs);
 
     int doc_len;
-    const void *doc = astarte_bson_serializer_get_document(&bs, &doc_len);
+    const void *doc = astarte_bson_serializer_get_document(bs, &doc_len);
```
Comment (sorru94) [inline src/edgehog_ota.c:576]: `doc_len` is not required as an argument. You can pass NULL.

File: src/edgehog_storage_usage.c [hunk 4]
```diff
@@ -78,8 +77,8 @@ static void publish_storage_usage(
     snprintf(path, path_size, "/%s", label);
 
     int doc_len;
-    const void *doc = astarte_bson_serializer_get_document(&bs, &doc_len);
+    const void *doc = astarte_bson_serializer_get_document(bs, &doc_len);
```
Comment (sorru94) [inline src/edgehog_storage_usage.c:80]: `doc_len` is not required as an argument. You can pass NULL.

File: src/edgehog_ota.c (hunk 1 above)
Comment (harlem88) [reply src/edgehog_ota.c:216]: It used here https://github.com/edgehog-device-manager/edgehog-esp32-device/blob/b7b626035d41d89cceddb97e071a3bd0c2f4ae66/src/edgehog_ota.c#L248

File: src/edgehog_ota.c (hunk 2 above)
Comment (harlem88) [reply src/edgehog_ota.c:229]: It used here https://github.com/edgehog-device-manager/edgehog-esp32-device/blob/b7b626035d41d89cceddb97e071a3bd0c2f4ae66/src/edgehog_ota.c#L250

**Review by sorru94 — APPROVED** [review]

**Pull request merged**